internal/initializer/crds.go

```diff
@@ -131,6 +134,45 @@ func (c *CoreCRDs) Run(ctx context.Context, kube client.Client) error {
 		if err := pa.Apply(ctx, crd); err != nil {
 			return errors.Wrap(err, "cannot apply crd")
 		}
+
+		crdNames = append(crdNames, crd.Name)
+	}
+
+	for _, crdName := range crdNames {
+		if err := waitForCRDEstablished(ctx, kube, crdName); err != nil {
+			return errors.Wrapf(err, "CRD %s not established", crdName)
+		}
 	}
```

Comment:
this means we'll wait 1 minute for each crd we are applying, not sure how many crds we have today, but could potentially be too long. Maybe you could make it `waitForCRDsEstablished`, pass the `crdNames` to it and with a single 60 seconds timeout, you could also just using `context.WithTimeout`, check all the missing crds each time, removing the ones you find established at each round, and exiting if all are established.

Reply:
Sure, we can do that, and it also came to my mind.

I did this with the point keeping in mind that if even one of the CRD is established(in this case first one), others will also get established within similar time frame as they were already registered in previous loop and we won't be waiting for 1 minute for each CRD. And with my approach, it will allow CRD to keep growing indefinitely without changing this waiting period. If we will keep a common waiting period of 1 minute for all CRDs, it can happen someday you'll have to change this limit.

Also, If you want this to be fast, we can use go routines for each CRD and then eventually wait for all of them to be finished. I was resisting it to not overwhelm k8s api server with all these concurrent requests.

LMK if you feel it needs to be changed and I'll update the PR accordingly. Let's close this long pending issue. It's always feels good to clear backlog issues😉